src/azul/drs.py

```diff
@@ -74,10 +87,164 @@ class Access:
     headers: Optional[Mapping[str, str]] = None
 
 
+@attr.s()
+class DRSURI(metaclass=ABCMeta):
+
+    @property
+    @abstractmethod
+    def scheme(self) -> str:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def object_id(self) -> str:
```

Comment:
This should be protected. And please make it a habit to provide a docstring for all abstract methods.